system/churchdb/upload.php

```diff
@@ -144,10 +144,14 @@ function handleUpload($uploadDirectory, $replaceOldFile = FALSE){
 
         if ($this->file->save($uploadDirectory . $filename . '.' . $ext)){
 
-
-          $person_id = $_GET['userid'];
-          $filename_old=$uploadDirectory . $filename . '.' . $ext;
-          $filename=md5(uniqid()).'.'.$ext;
+          $filename_old = $uploadDirectory . $filename . '.' . $ext;
+            
+          if (function_exists('openssl_random_pseudo_bytes')) {
+              $filename = bin2hex(openssl_random_pseudo_bytes(32)).'.'.$ext;
+          } else {
+              $rands = array_map('mt_rand', array_fill(0, 10, 0), array_fill(0, 10, mt_getrandmax()));
+              $filename = md5(implode($rands)).'.'.$ext;
```

Comment:
SHA256 is way too long for the database column as you defined it ;)
And your point is also pointless, because we're not looking for collision resistance but rather for a way to come up with something that cannot be guessed, and the 128 bits of MD5 are perfectly fine here. ;)